contracts/libraries/GnosisAuction.sol

```diff
@@ -103,6 +123,73 @@ library GnosisAuction {
         );
     }
 
+    function placeBid(BidDetails memory bidDetails)
+        internal
+        returns (
+            uint96 sellAmount,
+            uint96 buyAmount,
+            uint64 userId
+        )
+    {
+        // calculate how much to allocate
+        sellAmount = uint96(
+            bidDetails.lockedBalance.mul(bidDetails.optionAllocationPct).div(
+                10000
+            )
+        );
+
+        // divide the `asset` sellAmount by the target premium per oToken to
+        // get the number of oTokens to buy (8 decimals)
+        buyAmount = uint96(
+            uint256(sellAmount)
+                .mul(10**bidDetails.assetDecimals)
+                .div(bidDetails.optionPremium)
+                .mul(10**8)
+                .div(10**bidDetails.assetDecimals)
```

Comment:
multiplying by asset decimals twice - once for bidDetails.optionPremium which has `asset` decimals and once for shifting it to 8 decimals